docs/source/debugging.rst

```diff
@@ -1,6 +1,120 @@
 Troubleshooting and Debugging
 =============================
 
+Common Troubleshooting Scenarios
+--------------------------------
+
+This section includes common problems you might encounter while using the
+platform and how you can troubleshoot and debug them.
+
+Rule is not being matched / action is not firing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You created a rule which supposed to fire an action when a particular trigger
+is emitted, but for some reason, this action is not being called.
+
+There are multiple reasons why this could be the case:
+
+1. Trigger is not being emitted
+2. Rule criteria doesn't match the emitted trigger
+3. Action runner service is not running
+4. Combination of the above
+
+In this section we will guide you through troubleshooting this issue.
+
+1. Checking that the trigger is emitted
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First thing you should do is check that trigger is being emitted and flowing
+to the rules engine.
+
+The easiest way to do that is to check the rules_engine service log.
+
+1. Check the rules engine service log
+
+By default, rules_engine logs are stored in ``/var/log/st2rulesengine.log``
+file. You should inspect this file (``cat``, ``grep`` and ``tail`` are your
+friends) and look for a line similar to the one bellow.
+
+.. code-block:: bash
+
+    2015-02-23 15:13:51,250 INFO [-] Found <n> rules defined for trigger <trigger name>
+
+For example:
+
+.. code-block:: bash
+
+    2015-02-23 15:13:51,250 INFO [-] Found 1 rules defined for trigger st2.generic.actiontrigger
+
+2. Checking the sensor container log
+
+As an alternative, you can check the sensor container service log
+(``/var/log/st2/st2sensorcontainer.*.log``). Keep in mind that this approach
+only works for triggers being emitted by sensors and won't work for triggers
+generated by incoming webhooks and triggers generated by timers.
+
+If you don't see this line, this means that trigger is not being emitted and
+flowing to the rules_engine. This could either mean that the sensor is
+misconfigured or not running or that there is some other issue with the sensor.
```

Comment:
We also need a section on how to debug sensors either using sensor container --sensor-name option or using the sensor wrapper directly, but, baby steps.